ra/ra.go

```diff
@@ -685,6 +714,42 @@ func (ra *RegistrationAuthorityImpl) checkAuthorizations(ctx context.Context, na
 			strings.Join(badNames, ", "),
 		)
 	}
+
+	return nil
+}
+
+func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, names []string) error {
+	wg := sync.WaitGroup{}
+	ch := make(chan *probs.ProblemDetails, len(names))
+	for _, name := range names {
+		wg.Add(1)
+		go func(name string) {
+			resp, err := ra.caa.IsCAAValid(ctx, &vaPB.IsCAAValidRequest{
+				Domain: &name,
+			})
+			if err != nil {
+				ra.log.AuditErr(fmt.Sprintf("Rechecking CAA: %s", err))
+				ch <- probs.ServerInternal("Internal error rechecking CAA for " + name)
+			} else if resp.Problem != nil {
+				ch <- &probs.ProblemDetails{
+					Type:   probs.ProblemType(*resp.Problem.ProblemType),
+					Detail: *resp.Problem.Detail,
+				}
+			}
+			wg.Done()
+		}(name)
+	}
+	wg.Wait()
+	close(ch)
+	var fails []*probs.ProblemDetails
+	for err := range ch {
+		if err != nil {
+			fails = append(fails, err)
+		}
+	}
+	if len(fails) > 0 {
```

Comment:
Since we fail on any errors is there any benefit to failing out early here? (i.e. instead of waiting for the `sync.WaitGroup` to be done just ranging over the error channel and if anything comes through failing out then)